vendor/raygui/raygui.odin

```diff
@@ -0,0 +1,555 @@
+    SLIDER_WIDTH = 16,          // Slider size of internal bar
+    SLIDER_PADDING,             // Slider/SliderBar internal bar padding
+}
+
+// ProgressBar
+GuiProgressBarProperty :: enum c.int {
+    PROGRESS_PADDING = 16,      // ProgressBar internal padding
+}
+
+// ScrollBar
+GuiScrollBarProperty :: enum c.int {
+    ARROWS_SIZE = 16,
+    ARROWS_VISIBLE,
+    SCROLL_SLIDER_PADDING,      // (SLIDERBAR, SLIDER_PADDING)
+    SCROLL_SLIDER_SIZE,
+    SCROLL_PADDING,
+    SCROLL_SPEED,
+}
+
+// CheckBox
+GuiCheckBoxProperty :: enum c.int {
+    CHECK_PADDING = 16,          // CheckBox internal check padding
+}
+
+// ComboBox
+GuiComboBoxProperty :: enum c.int {
+    COMBO_BUTTON_WIDTH = 16,    // ComboBox right button width
+    COMBO_BUTTON_SPACING,       // ComboBox button separation
+}
+
+// DropdownBox
+GuiDropdownBoxProperty :: enum c.int {
+    ARROW_PADDING = 16,         // DropdownBox arrow separation from border and items
+    DROPDOWN_ITEMS_SPACING,     // DropdownBox items separation
+}
+
+// TextBox/TextBoxMulti/ValueBox/Spinner
+GuiTextBoxProperty :: enum c.int {
+    TEXT_INNER_PADDING = 16,    // TextBox/TextBoxMulti/ValueBox/Spinner inner text padding
+    TEXT_LINES_SPACING,         // TextBoxMulti lines separation
+    TEXT_ALIGNMENT_VERTICAL,    // TextBoxMulti vertical alignment: 0-CENTERED, 1-UP, 2-DOWN
+    TEXT_MULTILINE,             // TextBox supports multiple lines
+    TEXT_WRAP_MODE,             // TextBox wrap mode for multiline: 0-NO_WRAP, 1-CHAR_WRAP, 2-WORD_WRAP
+}
+
+// Spinner
+GuiSpinnerProperty :: enum c.int {
+    SPIN_BUTTON_WIDTH = 16,     // Spinner left/right buttons width
+    SPIN_BUTTON_SPACING,        // Spinner buttons separation
+}
+
+// ListView
+GuiListViewProperty :: enum c.int {
+    LIST_ITEMS_HEIGHT = 16,     // ListView items height
+    LIST_ITEMS_SPACING,         // ListView items separation
+    SCROLLBAR_WIDTH,            // ListView scrollbar size (usually width)
+    SCROLLBAR_SIDE,             // ListView scrollbar side (0-left, 1-right)
+}
+
+// ColorPicker
+GuiColorPickerProperty :: enum c.int {
+    COLOR_SELECTOR_SIZE = 16,
+    HUEBAR_WIDTH,               // ColorPicker right hue bar width
+    HUEBAR_PADDING,             // ColorPicker right hue bar separation from panel
+    HUEBAR_SELECTOR_HEIGHT,     // ColorPicker right hue bar selector height
+    HUEBAR_SELECTOR_OVERFLOW,   // ColorPicker right hue bar selector overflow
+}
+
+SCROLLBAR_LEFT_SIDE :: 0
+SCROLLBAR_RIGHT_SIDE :: 1
+
+//----------------------------------------------------------------------------------
+// Global Variables Definition
+//----------------------------------------------------------------------------------
+// ...
+
+//----------------------------------------------------------------------------------
+// Module Functions Declaration
+//----------------------------------------------------------------------------------
+
+@(default_calling_convention="c")
+foreign lib {
```

Comment:
As every procedure is prefixed with `Gui` in this package already, it might be better to remove the prefix entirely.

```odin
@(link_prefix="Gui")
```

Reply:
Does that make all the functions be prefixed with Gui automagically?

Reply:
This one I didn't quite get. I tried adding @(link_prefix="Gui") below @(default_calling_convention="c"), but things didn't actually work as I expected. SoI left this one untouched. As I understood, this would entail removing "Gui" from all procedure names, but if we use raylib's and raygui's meta information to generate the bindings automatically in the future, this is only gonna make the parser more complicated unneceraly I suppose.